src/globus_sdk/services/auth/client/base.py

```diff
@@ -203,6 +197,106 @@ def _convert_listarg(
+        :type query_params: dict, optional
+
+        .. tab-set::
+
+            .. tab-item:: Example Usage
+
+                .. code-block:: pycon
+
+                    >>> ac = globus_sdk.AuthClient(...)
+                    >>> # get by ID
+                    >>> r = ac.get_identity_providers(ids="41143743-f3c8-4d60-bbdb-eeecaba85bd9")
+                    >>> r.data
+                    {
+                      "identity_providers": [
+                        {
+                          "alternative_names": [],
+                          "name": "Globus ID",
+                          "domains": ["globusid.org"],
+                          "id": "41143743-f3c8-4d60-bbdb-eeecaba85bd9",
+                          "short_name": "globusid"
+                        }
+                      ]
+                    }
+                    >>> ac.get_identities(
+                    ...     ids=["41143743-f3c8-4d60-bbdb-eeecaba85bd9", "927d7238-f917-4eb2-9ace-c523fa9ba34e"]
+                    ... )
+                    >>> # or by domain
+                    >>> ac.get_identities(domains="globusid.org")
+                    >>> ac.get_identities(domains=["globus.org", "globusid.org"])
+
+                The result itself is iterable, so you can use it like so:
+
+                .. code-block:: python
+
+                    for idp in ac.get_identity_providers(domains=["globus.org", "globusid.org"]):
+                        print(f"name: {idp['name']}")
+                        print(f"id: {idp['id']}")
+                        print(f"domains: {idp['domains']}")
+                        print()
+
+            .. tab-item:: Example Response Data
+
+                .. expandtestfixture:: auth.get_identity_providers
+
+            .. tab-item:: API Info
+
+                ``GET /v2/api/identity_providers``
+
+                .. extdoclink:: Get Identity Providers
+                    :ref: auth/reference/#get_identity_providers
+        """  # noqa: E501
+
+        log.info("Looking up Globus Auth Identity Providers")
+
+        if query_params is None:
+            query_params = {}
+
+        if domains is not None and ids is not None:
+            raise exc.GlobusSDKUsageError(
+                "AuthClient.get_identity_providers does not take both "
+                "'domains' and 'ids'. These are mutually exclusive."
+            )
+        # if either of these params has a truthy value, stringify it
+        # this handles lists of values as well as individual values gracefully
+        # letting us consume args whose `__str__` methods produce "the right
+        # thing"
+        elif domains:
+            query_params["domains"] = _commasep(domains)
+        elif ids:
+            query_params["ids"] = _commasep(ids)
+        else:
+            log.warning(
+                "neither 'domains' nor 'ids' provided to get_identity_providers(). "
+                "This can only succeed if 'query_params' were given."
```

Comment:
This is technically not accurate -- reaching this point in the code indicates that one *was provided* but was an empty value.

```suggestion
                "'domains' or 'ids' was an empty value in get_identity_providers(). "
                "The call can only succeed if 'query_params' were given."
```

Reply:
My intent was to handle the case in which neither `domains` nor `ids` has been passed.
e.g.
```python
client.get_identity_providers(query_params={"other": "paramval"})
```

I see that you're calling out `ids=[]` as a possible usage which would still reach this.

Before I take this suggestion, which is one possible fix, what do we think of being more pedantic in our matching by switching the above to this?
```python
elif domains is not None: ...
elif ids is not None: ...
else: ...
```

It would mean that `ids=[]` gets passed through as a query param, but I think that's fine.

Reply:
I've gone ahead with a change to tighten these checks to `is not None`.
Let me know if you like (or dislike!) the results, sans any change to the warning log.